lib/analyzer/image-extractor.ts

```diff
@@ -1,56 +1,61 @@
 import { createReadStream } from "fs";
-import * as md5 from "md5";
 import * as minimatch from "minimatch";
 import { basename } from "path";
 import { Readable } from "stream";
 import { extract, Extract } from "tar-stream";
-import { streamToBuffer, streamToString } from "../stream-utils";
+import { streamToString } from "../stream-utils";
 
-export { extractImageKeyFiles, ExtractedImage, ExtractedKeyFiles };
+export {
+  extractFromTar,
+  ExtractedFiles,
+  ExtractFileCallback,
+  mapLookups,
+  LookupEntry,
+};
 
-interface ExtractedKeyFiles {
-  txt: { [key: string]: string };
-  md5: { [key: string]: string };
+interface ExtractedFiles {
+  [key: string]: string;
 }
 
-interface ExtractedImage {
-  manifest: string;
-  layers: Layers;
+interface ExtractedFilesByLayers {
+  [key: string]: ExtractedFiles;
 }
 
-interface Layers {
-  [key: string]: ExtractedKeyFiles;
+type ExtractFileCallback = (stream: Readable) => Promise<string>;
+
+interface LookupEntry {
```

Comment:
I think Lookup isn't a good name - the name of the interface should explain the it's related to callbacks